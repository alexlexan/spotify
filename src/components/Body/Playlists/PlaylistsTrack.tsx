import React from "react";
import "./Playlist.sass";
import { useDispatch } from "react-redux";
import { playAndSetTrack, setSongs } from "../../../store/actions";
import { PlaylistsTrackType, Track } from "../../../store/reducer";

type Props = {
  playlists: PlaylistsTrackType;
  title: string;
};

const PlaylistsTrack: React.FC<Props> = ({ playlists, title }) => {
  const dispatch = useDispatch();
  const play = (track: Track) => {
    dispatch(playAndSetTrack(track));
    dispatch(setSongs(null));
  };

  if (!playlists?.items.length) {
    return null;
  }

  return (
    <div className="playlist">
      <h1>{title}</h1>
      <div className="grid">
        {playlists?.items.map((item, index) => (
          <div
            className="grid__item playlist__item"
            key={index}
            onClick={() => play(item)}
          >
            <img src={item.album.images[0]?.url} alt={item.name} />
            <p>{item.name}</p>
          </div>
        ))}
      </div>
    </div>
  );
};

export default PlaylistsTrack;
