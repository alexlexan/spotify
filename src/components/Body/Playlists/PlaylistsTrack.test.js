import React from "react";
import PlaylistsTrack from "./PlaylistsTrack";

const mockDispatch = jest.fn();
jest.mock("react-redux", () => ({
  ...require.requireActual("react-redux"),
  useDispatch: () => mockDispatch,
}));

describe("tests PlaylistsTrack component with props", () => {
  const firstTrack = {
    album: {
      images: [{ url: "firstTrackUrl" }],
    },
    name: "firstTrack",
  };
  const secondTrack = {
    name: "secondTrack",
    album: {
      images: [{ url: "secondTrackUrl" }],
    },
  };
  const playlists = {
    items: [firstTrack, secondTrack],
  };

  afterEach(() => {
    mockDispatch.mockClear();
  });

  it("should render PlaylistsTrack with props", () => {
    const component = shallow(
      <PlaylistsTrack playlists={playlists} title={"test"} />
    );
    expect(component).toMatchSnapshot();
  });

  it("if click item should call dispatch 2 times", () => {
    const component = shallow(
      <PlaylistsTrack playlists={playlists} title={"test"} />
    );
    component
      .find(".playlist__item")
      .first()
      .simulate("click", playlists.items[0]);
    expect(mockDispatch).toBeCalledTimes(2);
  });
});

describe("tests PlaylistsTrack component without props", () => {
  it("should render PlaylistsTrack without props", () => {
    const component = shallow(<PlaylistsTrack />);
    expect(component).toMatchSnapshot();
  });
});
