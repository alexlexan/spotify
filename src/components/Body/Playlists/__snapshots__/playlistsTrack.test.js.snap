// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests PlaylistsTrack component with props should render PlaylistsTrack with props 1`] = `
<div
  className="playlist"
>
  <h1>
    test
  </h1>
  <div
    className="grid"
  >
    <div
      className="grid__item playlist__item"
      key="0"
      onClick={[Function]}
    >
      <img
        alt="firstTrack"
        src="firstTrackUrl"
      />
      <p>
        firstTrack
      </p>
    </div>
    <div
      className="grid__item playlist__item"
      key="1"
      onClick={[Function]}
    >
      <img
        alt="secondTrack"
        src="secondTrackUrl"
      />
      <p>
        secondTrack
      </p>
    </div>
  </div>
</div>
`;

exports[`tests PlaylistsTrack component without props should render PlaylistsTrack without props 1`] = `""`;
